### PR TITLE
Support for assigning nicknames to other users

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

File: backend/canisters/user/api/src/updates/set_contact.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    pub nickname: OptionUpdate<String>,
+}
+
+#[derive(CandidType, Serialize, Deserialize, Debug)]
+pub enum Response {
+    Success,
+    NoChange,
+    NicknameTooShort(FieldTooShortResult),
+    NicknameTooLong(FieldTooLongResult),
+    NicknameNameAlreadyUsed,
```
Comment (hpeebles) [inline backend/canisters/user/api/src/updates/set_contact.rs:23]: This can go

File: backend/canisters/user/impl/src/model/contacts.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    NicknameTooShort(FieldTooShortResult),
+    NicknameTooLong(FieldTooLongResult),
+}
+
+impl Contacts {
+    pub fn set_contact(&mut self, contact: OptionalContact) -> SetContactResponse {
+        match contact.nickname {
+            OptionUpdate::NoChange => SetContactResponse::NoChange,
+            OptionUpdate::SetToNone => {
+                if self.map.remove(&contact.user_id).is_some() {
```
Comment (hpeebles) [inline backend/canisters/user/impl/src/model/contacts.rs:32]: We need to be careful here that when we add more fields we only remove contacts which are empty.

File: backend/canisters/user/api/src/updates/set_contact.rs (hunk 1 above)
Comment (megrogan) [reply backend/canisters/user/api/src/updates/set_contact.rs:23]: Unsaved change in IDE!

File: backend/canisters/user/impl/src/model/contacts.rs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+                if length_provided < MIN_NICKNAME_LEN {
+                    return SetContactResponse::NicknameTooShort(FieldTooShortResult {
+                        length_provided,
+                        min_length: MIN_NICKNAME_LEN,
+                    });
+                }
+
+                self.map
+                    .entry(contact.user_id)
```
Comment (hpeebles) [inline backend/canisters/user/impl/src/model/contacts.rs:57]: You can do `entry.or_default().nickname = Some(nickname);`

**Review by hpeebles — APPROVED** [review]

**Pull request merged**